src/ctx.cpp

```diff
@@ -3868,7 +3868,7 @@ llvm::Value *FunctionEmitContext::XeSimdCFAny(llvm::Value *value) {
 }
 
 llvm::Value *FunctionEmitContext::XeSimdCFPredicate(llvm::Value *value, llvm::Value *defaults) {
-    AssertPos(currentPos, llvm::isa<llvm::VectorType>(value->getType()));
+    AssertPos(currentPos, llvm::isa<llvm::FixedVectorType>(value->getType()));
     llvm::FixedVectorType *vt = llvm::dyn_cast<llvm::FixedVectorType>(value->getType());
     if (defaults == nullptr) {
```

Comment:
Same issue as below: https://github.com/ispc/ispc/pull/2610/files#r1296438132, but here, `vt` is not only used during during a `Debug` build, so do the check in the guard to satisfy the analyzer.

```suggestion
    AssertPos(currentPos, llvm::isa<llvm::FixedVectorType>(value->getType()));
    llvm::FixedVectorType *vt = llvm::dyn_cast<llvm::FixedVectorType>(value->getType());
    if (defaults == nullptr && vt != nullptr) {
```